core/src/main/java/hudson/model/AbstractItem.java

```diff
@@ -351,6 +357,11 @@ protected void checkRename(@Nonnull String newName) throws Failure {
      * you can use this method.
      */
     protected void renameTo(final String newName) throws IOException {
+
+        if (!isNameEditable()) {
+            throw new IllegalArgumentException("Trying to rename an item that has not editable name.");
```

Comment:
Please throw IOException which is a checked exception in API. Throwing an undocumented runtime exception may cause issues for API consumers

Reply:
done.